test/extensions/tracers/datadog/agent_http_client_test.cc

```diff
@@ -564,7 +579,8 @@ TEST_F(DatadogAgentHttpClientTest, SkipReportIfCollectorClusterHasBeenRemoved) {
 
     // Attempt to send a request.
     const auto ignore = [](auto&&...) {};
-    datadog::tracing::Expected<void> result = client_.post(url_, ignore, "", ignore, ignore);
+    datadog::tracing::Expected<void> result = client_.post(
+        url_, ignore, "", ignore, ignore, time_.monotonicTime() + std::chrono::seconds(1));
```

Comment:
Please add a `TEST_F` for timeouts, if the mocking machinery permits it.

Regardless of whether we have a dedicated unit test, let's find out what timeouts actually do here within Envoy.